src/community/script/core/src/main/java/org/geoserver/script/rest/converter/ScriptConverter.java

```diff
@@ -0,0 +1,172 @@
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+
+/**
+ * This class enables us to extend the XMLXStreamListConverter and JSONXStreamListConverter classes so we can
+ * override the default XML and JSON output of the REST module.  Because these message converters have a lower
+ * priority than the converters in the REST module, they will always be called and therefore to prevent affecting
+ * the expected behavior of the REST module we are checking the request path and only customizing the output
+ * if the request is within the Scripts module.
+ */
+public abstract class ScriptConverter extends XStreamCatalogListConverter {
+
+    /**
+     * XML handling for Script lists
+     */
+    @Component
+    public static class XMLXStreamScriptListConverter extends XMLXStreamListConverter {
+        @Autowired
+        HttpServletRequest request;
+
+        @Override
+        protected XStream createXStreamInstance() {
+            return new SecureXStream();
+        }
+
+        @Override
+        public void encodeLink(String link, HierarchicalStreamWriter writer) {
+            String path = request.getRequestURI();
+            if (path.contains("script")) {
+                try {
+                    link = URLDecoder.decode(link, "UTF-8");
+                    encodeAlternateAtomLinkNoExt(link, writer);
+                    writer.addAttribute("type", getMediaType());
+
+                    writer.endNode();
+                } catch (UnsupportedEncodingException e) {
+
+                }
+            } else {
+                encodeAlternateAtomLink(link, writer);
+            }
+        }
+
+        @Override
+        public void encodeCollectionLink(String link, HierarchicalStreamWriter writer) {
+            String path = request.getRequestURI();
+            if (path.contains("script")) {
+                try {
+                    link = URLDecoder.decode(link, "UTF-8");
+                    encodeAlternateAtomLinkNoExt(link, writer);
+                    writer.addAttribute("type", getMediaType());
+
+                    writer.endNode();
+                } catch (UnsupportedEncodingException e) {
+
+                }
+            } else {
+                encodeAlternateAtomLink(link, writer);
+            }
+        }
+
+        @Override
+        public String getMediaType() {
+            return MediaType.APPLICATION_ATOM_XML_VALUE;
+        }
+
+        @Override
+        public String getExtension() {
+            return "xml";
+        }
+    }
+
+    /**
+     * JSON handling for Script lists
+     */
+    @Component
+    public static class JSONXStreamScriptListConverter extends JSONXStreamListConverter {
+        @Autowired
+        HttpServletRequest request;
+
+        @Override
+        public void encodeLink(String link, HierarchicalStreamWriter writer) {
+            String path = request.getRequestURI();
+            if (path.contains("script")) {
```

Comment:
Stricter check. Given that this is in 3 places, is there a better method that we can override that would only require a single path check?

Reply:
See comment containing new method above to facilitate this check.